accounts-db/src/accounts_index/bucket_map_holder.rs

```diff
@@ -207,9 +258,44 @@ impl<T: IndexValue, U: DiskIndexValue + From<T> + Into<T>> BucketMapHolder<T, U>
             .and_then(|drives| drives.first())
             .map(|drive| drive.join("accounts_index_restart"));
 
-        let disk = match config.index_limit_mb {
-            IndexLimitMb::InMemOnly => None,
-            IndexLimitMb::Minimal => Some(BucketMap::new(bucket_config)),
+        let disk = match config.index_limit {
+            IndexLimit::InMemOnly => None,
+            IndexLimit::Minimal | IndexLimit::Threshold(_) => Some(BucketMap::new(bucket_config)),
+        };
+
+        // Compute threshold_entries once here
+        let threshold_entries_per_bin = match config.index_limit {
+            IndexLimit::Threshold(limit_bytes) => {
+                let size_of_uninitialized = InMemAccountsIndex::<T, U>::size_of_uninitialized(); // 32 (pubkey) + 8 (box pointer) = 40
+                let size_of_single_entry = InMemAccountsIndex::<T, U>::size_of_single_entry(); // 48 (AccountMapEntry<T>)
+
+                // Calculate bytes allocated per bin
+                let bytes_per_bin = (limit_bytes as usize) / bins;
+
+                // Calculate entries per bin accounting for load factor:
+                // At 7/8 load factor, each entry effectively uses:
+                // (7/8 * full_entry_size) + (1/8 * uninitialized_size)
+                let full_entry_size = size_of_uninitialized + size_of_single_entry;
+                let weighted_bytes_per_entry =
+                    (7 * full_entry_size + size_of_uninitialized) / 8;
+                let entries_per_bin = bytes_per_bin / weighted_bytes_per_entry;
+
+                // Calculate threshold with log2 roundup and load factor (7/8)
+                let threshold_per_bin = if entries_per_bin == 0 {
+                    0
+                } else {
+                    let entries_per_bin_log2 = entries_per_bin.ilog2();
+                    (1usize << entries_per_bin_log2) * 7 / 8
+                };
+
+                info!(
+                    "AccountsIndex threshold configured: limit_bytes={limit_bytes}, bytes_per_bin={bytes_per_bin}, \
+                     entries_per_bin={entries_per_bin}, threshold_entries_per_bin={threshold_per_bin}, \
+                     size_of_uninitialized={size_of_uninitialized}, size_of_single_entry={size_of_single_entry}"
+                );
+                Some(threshold_per_bin)
+            }
+            _ => None,
```

Comment:
Can we not use the wildcard match, and instead enumerate all the variants? One more safe guard in case we modify the enum later. (And can the match arms for InMem & Minimal both be at the top, please, since they are one-liners?)